src/main/java/build/buildfarm/cas/cfc/CASFileCache.java

```diff
@@ -2819,23 +2822,38 @@ void commit() throws IOException {
           throw e;
         }
 
-        Entry entry = new Entry(key, blobSizeInBytes, Deadline.after(10, SECONDS));
+        // Create the new entry to track the file.
+        // FIXME: How is this deadline decided?
+        // FIXME: Could it get deleted before we finish committing?
+        // TEST: Bump it higher to see if it avoids failures setting up the exec dir.
+        Entry entry = new Entry(key, blobSizeInBytes, Deadline.after(5, MINUTES));
 
-        Entry existingEntry = null;
+        Entry existingEntry;
         boolean inserted = false;
         try {
           Files.createLink(CASFileCache.this.getPath(key), writePath);
           existingEntry = storage.putIfAbsent(key, entry);
           inserted = existingEntry == null;
         } catch (FileAlreadyExistsException e) {
-          log.log(Level.FINE, "file already exists for " + key + ", nonexistent entry will fail");
+          // FIXME: Why skip adding the entry to storage if the hardlink already exists?
+          // This seems like it would lead to the in-memory and disk storage being out of sync.
```

Comment:
If it already exists, it means that we lost the race to insert it, and another write must have successfully created the target link, and that it is now responsible for making both the storage entry and the file appear.